### PR TITLE
Adding scheduler notification

### PR DESCRIPTION
**Pull request by msalemcode**
Enable Scheduler engine to send email notification to ISV for missing, successful and failure metered event.
Also, logging the engine events into App logging table

### CODE REVIEW

File: src/Services/Helpers/EmailHelper.cs [hunk 1]
```diff
@@ -121,4 +121,64 @@ public EmailContentModel PrepareEmailContent(Guid subscriptionID, Guid planGuId,
...
+    /// <param name="schedulerTaskName">scheduler Task Name.</param>
+    /// <param name="responseJson">response Json.</param>
+    /// <param name="subscriptionStatus">The subscription status.</param>
+    /// <returns>
+    /// Email Content Model.
+    /// </returns>
+    /// <exception cref="Exception">Error while sending an email, please check the configuration.
+    /// or
+    /// Error while sending an email, please check the configuration.</exception>
+    public EmailContentModel PrepareMeteredEmailContent(string schedulerTaskName, String subscriptionName, string subscriptionStatus, string responseJson)
```
Comment (code4clouds) [inline src/Services/Helpers/EmailHelper.cs:138]: PrepareMeteredEmailContent and PrepareEmailContent are similar

Comment (code4clouds) [reply src/Services/Helpers/EmailHelper.cs:138]: It seems like we need a base class for setting the defaul emailcontent properties and then extend for the specific usage.

**Review by code4clouds — CHANGES_REQUESTED** [review]
It seems like we need a base class for setting the default emailcontent properties and then extend for the specific usage on both PrepareMeteredEmailContent and PrepareEmailContent

File: src/Services/Helpers/EmailHelper.cs (hunk 1 above)
Comment (msalemcode) [reply src/Services/Helpers/EmailHelper.cs:115]: cleaned up the code and it is optimized now

**Review by code4clouds — APPROVED** [review]

**Pull request merged**